### PR TITLE
Add the ability to configure the sort order and to use inventory tweaks

### PR DESCRIPTION
**Pull request by IMarvinTPA**
Add the ability to configure the sort order and to use inventory tweaks's sorting.

https://github.com/mezz/JustEnoughItems/issues/549

### CODE REVIEW

**Comment by ChloeDawn** [comment]
Congrats, you added a hard dependency for a single "feature".

**Comment by adrwa** [comment]
Why have you added a dep for inventory tweaks to JEI of all the things to dep?

**Comment by IMarvinTPA** [comment]
So you can have the inventory panel on the right of the screen sort the items by how you have your inventory tweaks tree setup.  I'm working on a very large tree to sort basically everything.  It is nice having all of my iron/copper/whatever ingots grouped together.
I don't know enough to know how to make it a "soft" dependency if you mean for building it.  It runs just fine without Inventory Tweaks added to an actual run.

**Comment by TehNut** [comment]
JEI has no dependencies, compile or otherwise, other than Forge and Minecraft. I highly doubt this will be merged if it adds one. 

This feature should be accomplished by exposing new methods to the API and having them added via plugins. *Not* in the core of the mod.

File: src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java [hunk 1]
```diff
@@ -2,41 +2,281 @@
...
+                String name2 = o2.getDisplayName();
+                return name1.compareTo(name2);
+            }
+        });
+		
+		//Inventory Tweaks needs to be part of the build environment for this bit to work.
+		boolean	hasInvTweaks = false;
+		try {
+			hasInvTweaks = Loader.isModLoaded("inventorytweaks"); 
+		} catch (Exception ex) {
```
Comment (adrwa) [inline src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java:198]: Why is this a try catch? Loader.isModLoaded doesn't throw an exception

**Review by adrwa — CHANGES_REQUESTED** [review]

File: src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java (hunk 1 above)
Comment (IMarvinTPA) [reply src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java:198]: It does during the gradle build test phase.  I could not find a way to detect the test was running to not call isModLoaded.

**Comment by IMarvinTPA** [comment]
I wouldn't mind having an API to connect to, but it is beyond my comfort level with Java to build one.  Feel free to use this as a basis for one and I'll make a tiny mod to connect the two APIs.

**Comment by mezz** [comment]
Sorry but this feature is going to take a lot more work to do cleanly, which is why I haven't gotten around to it yet.

**Comment by IMarvinTPA** [comment]
Thank you for looking at it.  Having the inventory panel sort is a big help in developing a tree file.  There is no chest big enough and easy enough to work with to hold one of everything and it is hard to even fill it anyway.  I don't know if AE has added sorting by Inventory Tweaks back in 1.10 or later either.  It was there in 1.7.  At least now it should be easier for me to maintain my personal JEI jar.

**Pull request closed without merging**